pkg/meta/lib/future_extensions.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Annotations that developers can use to extend [Future] capabilities.
+library meta_future_extensions;
+
+/// Extensions that apply to futures.
+extension FutureExtension<T> on Future<T>? {
```

Comment:
This name could be more specific. You're likely never going to need to write it, but if you are, it's nicer if it is somewhat related to the content.

Maybe `MetaFutureExtension` or `FutureLintExtension`. Or Just `FutureLintTools`, extensions don't have to be named `Extension` (we do tend to name extensions as `Extension` and mixins as `Mixin` because it's both their declaration and their *role*, but it's not a rule that you must, if you can describe their role better with other words).

Reply:
I don't have any problem making the name more specific, but I find "Tools" to be even more general than "Extensions", so I prefer `FutureLintExtensions`.

Reply:
And about the library name? Should I change it to `lint_tools.dart`?

Reply:
I think the general style is to use the snake-case equivalent of the declaration's name when there's a single declaration in the file, so maybe `future_lint_extensions.dart`? It's kind of long, and I'm not convinced that "lint" is adding anything, either in the file name or the extension name, but I don't have a strong preference.

Reply:
Regarding sibling file name like `dart2js.dart` it could be worth to name it `linter.dart`. It will open the door to adding annotations that can be leveraged by the linter package.

Reply:
About the name, as there can be several extension on the same type, we could name the extension `SilentUnawaitedFuturesExtension`!

Reply:
I don't have any objection to `linter.dart`. It indicates that the contents are related to the linter, which is true.

> ... adding annotations that can be leveraged by the linter package.

I'm not sure what you have in mind, but so far we've chosen to make enforcement of annotations be enabled by default and hence hints, rather than lints.

Reply:
> I'm not sure what you have in mind

There are several issue in the linter issue tracker mentioning annotations to know if a lint should apply or not. This `linter.dart` lib could be the recipient where those annotations land.